axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceLineServiceImpl.java

```diff
@@ -740,4 +755,24 @@ public Map<String, Map<String, Object>> setScale(InvoiceLine invoiceLine, Invoic
 
     return attrsMap;
   }
+
+  @Override
+  public String computeInvoiceLineTaxLineSetDomain(int operationTypeSelect) {
+    LocalDate todayDate = LocalDate.now();
+    String domain = "(self.endDate > :date OR self.endDate IS NULL) ";
+    if (operationTypeSelect == 3 || operationTypeSelect == 4) {
+      domain += " AND self.tax.isNonDeductibleTax = false ";
+    }
+
+    try {
+      List<Long> taxLineIds =
+          JPA.em()
+              .createQuery(" select self.id from TaxLine self where " + domain, Long.class)
+              .setParameter("date", todayDate)
+              .getResultList();
+      return "self.id in (" + Joiner.on(",").join(taxLineIds) + ")";
```

Comment:
It is not necessary to compute a list of ids, returning the where clause should be enough, so you could simply return "(self.endDate > :__date__ OR self.endDate IS NULL) " or "(self.endDate > :__date__ OR self.endDate IS NULL)  AND self.tax.isNonDeductibleTax = false " depending on the operationTypeSelect